src/frequenz/client/common/pagination/pagination.py

```diff
@@ -0,0 +1,100 @@
+# License: MIT
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+"""Module to define the pagination used with the common client."""
+
+from __future__ import annotations  # required for constructor type hinting
+
+import logging
+from dataclasses import dataclass
+
+# pylint: disable=import-error, no-name-in-module
+from frequenz.api.common.v1.pagination_info_pb2 import (
+    PaginationInfo as PBPaginationInfo,
+)
+
+# pylint: disable=import-error, no-name-in-module
+from frequenz.api.common.v1.pagination_params_pb2 import (
+    PaginationParams as PBPaginationParams,
+)
+
+# Set up logging
+_logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class PaginationParams:
```

Comment:
Here comes another design question. Do we want a wrapper over `PaginationParams` at all? I envision this never being exposed to the user. The client should probably offer a method with an async iterable interface, like:

```py
async def list(self, page_size: int | None = None, page_token: str | None = None) -> AsyncIterator[Whatever]:
```

So users can do:

```py
async for whatever in client.list():
    ...
```

Then `list()` (or the async iterator returned by it) should handle pagination internally, pagination is basically a protocol optimization. Maybe there could be advanced cases where users want to use explicit pagination, but I think that should be a low level interface. Only if we want to offer this low-level interface I think it makes sense to wrap `PaginationParams`.